### PR TITLE
Fix hovercolor not working in global tag

### LINKED ISSUE #9557: Hypertext: <global> color/hovercolor does not affect <action> links
##### Minetest version

```
3c0d30964315ad9e2cba2c6ade1260d8b35293da
```

##### Summary
If you set a `<global>` with `color` and `hovercolor` in a `hypertext[]` formspec element, and add an `<action>` link, the color of the `<action>` link is NOT affected.
Only the normal (not linked) text will be colorized.

##### Steps to reproduce
- Grab `devtest`
- Start it
- `/test_formspec`
- Click the “Hypertext” tab
- This should render 2 `hypertext[]` elements: one top, one bottom

We only care about the bottom now. The bottom element tests `<global>`. It sets all possible attributes that `<global>` can have, including `color` and `hovercolor`.

Expected: The action link takes the color and hovercolor as set in `<global>`
Actual: Hyperlink color remains at default (blue / red).

Note: You can find the testing code in the `testformspec` mod.

### PR DESCRIPTION
**Pull request by pyrollo**
This fixes #9557, making hovercolor global tag attribute working as expected and as told in lua_api.txt. 

Anyway, I'm wondering if this is behavior is really adequate. There is a <tag> tag that allow to create new tags or redefine existing ones. So changing `action` color and hover color can be done with :

`<tag name=action hovercolor=blue color=red>`

I think hovercolor should be removed from `<global>` (or actually change hovercolor of all elements) in profit of `<tag>`.

Something can be improved in `<tag>`, if you use :

`<tag name=action hovercolor=blue>`

This resets "color" for `<action>` to default color (instead of the default specific actions color). Maybe it should only change mentioned styles on existing tags and leave other styles unchanged.

#### TL;DR:

This proposal: Fix global hovercolor.

Alternative proposal: Remove hovercolor from `<global>`, use `<tag name=action hovercolor=...>` instead.

## How to test
Test with formspec: 
`size[10,10]hypertext[0.2,1;10,9;test;<global hovercolor=green><action name=test>Test</action>"]`

Should display a "Test" link, when hovered should turn to green instead of default red.

### CODE REVIEW

**Comment by SmallJoker** [comment]
I don't think `hovercolor` fits into `<global ...>`. Tag types should be styled individually.

**Comment by rubenwardy** [comment]
I think that fixing the documented behaviour makes sense this close to release, and I don't think the documented behaviour is necessarily bad. Redefining the tag like that is a bit strange to me
